cpp/include/cuspatial/detail/range/multipolygon_range.cuh

```diff
@@ -118,6 +118,19 @@ multipolygon_range<GeometryIterator, PartIterator, RingIterator, VecIterator>::m
     num_points(), num_multipolygons() + 1, num_polygons() + 1, num_rings() + 1);
 }
 
+template <typename GeometryIterator,
+          typename PartIterator,
+          typename RingIterator,
+          typename VecIterator>
+CUSPATIAL_HOST_DEVICE constexpr bool
+multipolygon_range<GeometryIterator, PartIterator, RingIterator, VecIterator>::
+  contains_only_single_geometry()
+{
+  using index_t =
+    multipolygon_range<GeometryIterator, PartIterator, RingIterator, VecIterator>::index_t;
+  return std::is_same_v<GeometryIterator, thrust::counting_iterator<index_t>>;
+}
```

Comment:
We can actually check this rather cheaply at runtime. Simply check `num_polygon == num_multipolygon`. Will change this.

Reply:
Is there a need to do it at runtime rather than statically?

Reply:
Yes, because checking if the iterator is a `counting_iterator` is too strict a check for single-polygon range. You can have a materialized array that stores a sequence, each increment by 1 and still represent a single-polygon range.

Reply:
The static check can be named something else, "is_lazy_evaluated_single_polygon_range"? If we find a use to it.

Reply:
Just make it runtime and if we need a static version later we can add it.